x11-misc/xflux/xflux-20130927.ebuild

```diff
@@ -0,0 +1,36 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DESCRIPTION="Makes the color of your computer's display adapt to the time of the day"
+HOMEPAGE="https://justgetflux.com/"
+SRC_URI="amd64?  ( https://justgetflux.com/linux/${PN}64.tgz -> ${PN}64-${PV}.tar.gz )
+	x86?	( https://justgetflux.com/linux/${PN}-pre.tgz -> ${P}.tar.gz )"
+
+KEYWORDS="-* ~amd64 ~x86"
+LICENSE="f.flux"
+SLOT="0"
+
+RESTRICT="bindist mirror strip"
+
+RDEPEND="dev-libs/libbsd
+	sys-devel/gcc
```

Comment:
you should not depend on `sys-devel/gcc` itself

Reply:
Even, when ldd lists a depency?

```
╰─➤  ldd /usr/bin/xflux
        linux-vdso.so.1 (0x00007fff3e9db000)
        libXxf86vm.so.1 => /usr/lib64/libXxf86vm.so.1 (0x0000003bed200000)
        libXext.so.6 => /usr/lib64/libXext.so.6 (0x0000003beca00000)
        libX11.so.6 => /usr/lib64/libX11.so.6 (0x0000003bed600000)
        libXrandr.so.2 => /usr/lib64/libXrandr.so.2 (0x0000003bf5e00000)
        libstdc++.so.6 => /usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/libstdc++.so.6 (0x0000003beee00000)
        libm.so.6 => /lib64/libm.so.6 (0x0000003beaa00000)
        libgcc_s.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/libgcc_s.so.1 (0x0000003beba00000)
        libc.so.6 => /lib64/libc.so.6 (0x0000003bea600000)
        libxcb.so.1 => /usr/lib64/libxcb.so.1 (0x0000003bece00000)
        libdl.so.2 => /lib64/libdl.so.2 (0x0000003beb200000)
        libXrender.so.1 => /usr/lib64/libXrender.so.1 (0x0000003bec600000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fdb8bf2a000)
        libXau.so.6 => /usr/lib64/libXau.so.6 (0x0000003bee200000)
        libXdmcp.so.6 => /usr/lib64/libXdmcp.so.6 (0x0000003bede00000)
        libbsd.so.0 => /usr/lib64/libbsd.so.0 (0x0000003beda00000)
        librt.so.1 => /lib64/librt.so.1 (0x0000003bec200000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x0000003beae00000)
```

Reply:
`ldd` is wrong and should **never** be used for this purpose, because it shows inderect runtime dependencies among direct ones.

I would recommend you this script:
`qlist -e "$@" | xargs scanelf -L -n -q -F '%n #F' | tr , ' ' | xargs qfile -Cv | sort -u | awk '{print $1}' | uniq`
which can be put into a file as a shell script and then run like `./rdeps.sh <binary_name>`

But be carefull, it also shows the packages from the `@system` set, which never should be included as a runtime dependency, so make sure you checked `emerge -pv @system` output before make a conclusion about what should be added as a runtime dependency and what should not

Reply:
Looking at you script, I get this output:
```
╰─➤  ./scanportage xflux                                                                                                                                                                                 130 ↵
sys-devel/gcc-7.3.0
sys-libs/glibc-2.26-r6
x11-libs/libX11-1.6.5
x11-libs/libXext-1.3.3
x11-libs/libXrandr-1.5.1
x11-libs/libXxf86vm-1.1.4
```

Minus gcc and glibc, 4 deps are left.
So the others, like libXrandr.so.2, whould be indirect runtime deps?

Reply:
the rest are candidates to be your runtime dependencies, you should use clear gentoo installation (like chroot) to figure out what you actually need

Reply:
If this is a binary package, then dep on gcc/glibc can actually be valid since it doesn't use *any* compiler you might use.

Reply:
That's why you have to check and do not just trust ldd

Reply:
Thanks. I've pushed now a new version with the newer deps.